docs/root/intro/arch_overview/other_features/compression.rst

```diff
@@ -0,0 +1,20 @@
+.. _arch_overview_compression:
+
+Compression
+===========
+
+Underlying implementation
+-------------------------
+
+Currently Envoy is written to use `zlib <http://zlib.net>`_ as the compression provider.
+
+.. note::
+
+  `zlib-ng <https://github.com/zlib-ng/zlib-ng>`_ is a fork that hosts several 3rd-party
+  contributions containing new optimizations. Those optimizations are considered useful for
+  `improving compression performance <https://github.com/envoyproxy/envoy/issues/8448#issuecomment-667152013>`_.
+  Envoy can be built with `zlib-ng <https://github.com/zlib-ng/zlib-ng>`_ support by using
```

Comment:
There's one confusing thing here (to me):

Being built with zlib-ng implies using it -- right? To me this implies it still needs to be turned on at runtime, which doesn't seem to be the case.

Shall we say "be built to use zlib-ng instead of regular zlib" to be extra clear?